### PR TITLE
[Ready]Adds circuitry lab to Pubbystation

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
add: Pubbystation's RnD department has been outfitted with a brand new circuitry lab! It is found east of the Toxins launch room.
/:cl:

I replaced RnD's ancient mineral storage room with a fancy circuitry lab.

Let me know if you think something is missing.

autism-awareness-week

### CODE REVIEW

**Comment by Jordie0608** [comment]
I think you're missing some outside corners.

**Comment by arsserpentarium** [comment]
Cool lab.

**Comment by 81Denton** [comment]
@Jordie0608 Now that you mention it... Give me a few minutes.

**Comment by Mark9013100** [comment]
Could use some purple floor tiles, that room looks a little bland.

**Comment by ShizCalev** [comment]
.....
i go away for three days and WHAT THE FUCK HAPPENED TO SPACE

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 1]
```diff
@@ -50185,6 +50340,19 @@
 	},
 /turf/open/floor/plating,
 /area/ai_monitored/turret_protected/AIsatextAP)
+"loH" = (
+/obj/structure/table/reinforced,
+/obj/machinery/light{
+	dir = 1
+	},
+/obj/structure/sign/poster/contraband/busty_backdoor_xeno_babes_6{
+	pixel_x = 0;
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: dv

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 2]
```diff
@@ -50319,6 +50654,20 @@
...
+/obj/machinery/atmospherics/pipe/manifold/scrubbers/hidden{
+	dir = 8
+	},
+/obj/effect/turf_decal/stripes/line{
+	dir = 8
+	},
+/obj/machinery/light_switch{
+	dir = 8;
+	pixel_x = -20;
+	pixel_y = 0
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: dv

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: _maps/map_files/PubbyStation/PubbyStation.dmm (hunk 2 above)
Comment (81Denton) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: How do I fix them?

Comment (81Denton) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: @ShizCalev How do I fix dirty vars?

Comment (SpaceManiac) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: You have to edit them out in Notepad++ or another editor.

Comment (81Denton) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: So just remove the "pixel_y = 0" etc?

Comment (SpaceManiac) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: Yes, and make sure the previous line loses its semicolon.

**Comment by 81Denton** [comment]
Dirty vars are gone and I prettied up the floor a little.

**Comment by Bawhoppen** [comment]
Pubby is supposed to be a smaller station, right? This is the same size as other map's circuitry labs

**Comment by ShizCalev** [comment]
Yeah that's a pretty good point. Could probably do with shedding off one of the table/chair sections.

**Comment by 81Denton** [comment]
@Bawhoppen You'd be thinking of Omega; most of Pubby's departments are comparable to or just a bit smaller than Metastation's. The map just has a lot less maintenance and empty spaces.

For example, compare both maps' engineering, atmos, cargo, medbay and xeno sections.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 3]
```diff
@@ -50091,25 +50124,131 @@
...
+	dir = 8
+	},
+/area/science/circuit)
+"hsZ" = (
+/obj/structure/table/reinforced,
+/obj/item/stock_parts/cell/super,
+/obj/item/stock_parts/cell/super,
+/obj/machinery/computer/security/telescreen{
+	desc = "Used for watching the other eggheads from the comfort of the circuitry lab.";
+	dir = 1;
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: wrong direction here.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 4]
```diff
@@ -50166,6 +50324,28 @@
...
+	dir = 4
+	},
+/obj/effect/turf_decal/stripes/line,
+/turf/open/floor/plasteel,
+/area/science/circuit)
+"jOZ" = (
+/obj/structure/table/reinforced,
+/obj/machinery/computer/security/telescreen{
+	desc = "Used for watching the other eggheads from the comfort of the circuitry lab.";
+	dir = 1;
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: same

**Review by ShizCalev — DISMISSED** [review]

**Comment by pubby** [comment]
Rooms on the station's exterior can be as big as desired. There's really no concern for size there.

:+1: for this. One of these days I'll update the map move things around to fit better, but for now this is a big improvement.

**Review by ShizCalev — DISMISSED** [review]

**Comment by ShizCalev** [comment]
Wait a sec, lemme double check something.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 5]
```diff
@@ -34147,6 +34156,14 @@
...
+/obj/machinery/camera{
+	c_tag = "Toxins Launch Area";
+	dir = 2;
+	network = list("ss13","rd")
+	},
+/obj/structure/sign/poster/official/random{
+	pixel_y = 32
+	},
 /turf/open/floor/plasteel,
 /area/science/mineral_storeroom)
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: update the area please.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 6]
```diff
@@ -34207,18 +34230,20 @@
...
 	},
-/turf/open/floor/plasteel,
-/area/science/mineral_storeroom)
-"bFz" = (
+/turf/open/floor/plasteel/purple/side,
+/area/science/circuit)
+"bFy" = (
 /obj/machinery/power/apc{
 	dir = 1;
 	name = "Toxins Launch Room APC";
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: Incorrect name.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
No fire alarm in toxins launch area
No vent in toxins launch area.
Add a lightswitch in the toxins launch area.

**Comment by 81Denton** [comment]
@ShizCalev Ok, I changed what you requested.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 7]
```diff
@@ -34161,42 +34172,48 @@
 	pixel_y = 22
 	},
 /turf/open/floor/plasteel,
-/area/science/mineral_storeroom)
+/area/science/mixing)
 "bFu" = (
-/obj/structure/ore_box,
+/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden,
+/obj/machinery/power/apc{
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: Are there two APCs in this area now?

Comment (81Denton) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: Oops. I removed the duplicate APC and moved the air alarm to that tile.

File: _maps/map_files/PubbyStation/PubbyStation.dmm [hunk 8]
```diff
@@ -34268,18 +34303,30 @@
...
+	},
+/obj/structure/disposalpipe/segment{
+	dir = 4
+	},
+/obj/effect/mapping_helpers/airlock/cyclelink_helper{
+	dir = 8
+	},
 /turf/open/floor/plating,
-/area/maintenance/department/cargo)
+/area/science/circuit)
```
Comment (ShizCalev) [inline _maps/map_files/PubbyStation/PubbyStation.dmm]: /area/maintenance/department/cargo for the maint door's area.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: _maps/map_files/PubbyStation/PubbyStation.dmm (hunk 8 above)
Comment (81Denton) [reply _maps/map_files/PubbyStation/PubbyStation.dmm]: @ShizCalev Thanks for the reviews, I fixed the door's area.

**Comment by 81Denton** [comment]
@ShizCalev Sorry for the 325253 reviews you had to do. I'll check things more thoroughly next time.

**Pull request merged**